src/storage/hummock_sdk/src/version.rs

```diff
@@ -610,3 +883,1065 @@ impl HummockVersionDelta {
+        Self {
+            sst_infos: pb_validation_task
+                .sst_infos
+                .into_iter()
+                .map(SstableInfo::from)
+                .collect_vec(),
+            sst_id_to_worker_id: pb_validation_task.sst_id_to_worker_id.clone(),
+            epoch: pb_validation_task.epoch,
+        }
+    }
+}
+
+impl From<ValidationTask> for PbValidationTask {
+    fn from(validation_task: ValidationTask) -> Self {
+        Self {
+            sst_infos: validation_task
+                .sst_infos
+                .into_iter()
+                .map(|sst| sst.into())
+                .collect_vec(),
+            sst_id_to_worker_id: validation_task.sst_id_to_worker_id.clone(),
+            epoch: validation_task.epoch,
+        }
+    }
+}
+
+impl ValidationTask {
+    pub fn estimated_encode_len(&self) -> usize {
+        self.sst_infos
+            .iter()
+            .map(|sst| sst.estimated_encode_len())
+            .sum::<usize>()
+            + self.sst_id_to_worker_id.len() * (size_of::<u64>() + size_of::<u32>())
+            + size_of::<u64>()
+    }
+}
+
+#[derive(Clone, PartialEq, Default, Debug)]
+pub struct ReportTask {
+    pub table_stats_change: HashMap<u32, PbTableStats>,
+    pub task_id: u64,
+    pub task_status: TaskStatus,
+    pub sorted_output_ssts: Vec<SstableInfo>,
+}
+
+impl From<PbReportTask> for ReportTask {
+    fn from(value: PbReportTask) -> Self {
+        Self {
+            table_stats_change: value.table_stats_change.clone(),
+            task_id: value.task_id,
+            task_status: TaskStatus::try_from(value.task_status).unwrap(),
+            sorted_output_ssts: value
+                .sorted_output_ssts
+                .into_iter()
+                .map(SstableInfo::from)
+                .collect_vec(),
+        }
+    }
+}
+
+impl From<ReportTask> for PbReportTask {
+    fn from(value: ReportTask) -> Self {
+        Self {
+            table_stats_change: value.table_stats_change.clone(),
+            task_id: value.task_id,
+            task_status: value.task_status.into(),
+            sorted_output_ssts: value
+                .sorted_output_ssts
+                .into_iter()
+                .map(|sst| sst.into())
+                .collect_vec(),
+        }
+    }
+}
+
+// for version delta
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct IntraLevelDelta {
+    pub level_idx: u32,
+    pub l0_sub_level_id: u64,
+    pub removed_table_ids: Vec<u64>,
+    pub inserted_table_infos: Vec<SstableInfo>,
+    pub vnode_partition_count: u32,
+}
+
+impl IntraLevelDelta {
+    pub fn estimated_encode_len(&self) -> usize {
+        size_of::<u32>()
+            + size_of::<u64>()
+            + self.removed_table_ids.len() * size_of::<u32>()
+            + self
+                .inserted_table_infos
+                .iter()
+                .map(|sst| sst.estimated_encode_len())
+                .sum::<usize>()
+            + size_of::<u32>()
+    }
+}
```

Comment:
Can we split this file and move the new structs to separate files to avoid making this file too large?